### PR TITLE
Extract `cacheFrom` from original BuildImageConfiguration as a LIST instead of STRING

### PR DESCRIPTION
**Pull request by brenuart**
See #1455

### CODE REVIEW

**Comment by rohanKanojia** [comment]
I think you would need to update tests as well

**Comment by brenuart** [comment]
Yeah, I forgot :-(
I have some issues to run them under Eclipse.

**Comment by rohanKanojia** [comment]
Sorry I'm not an eclipse IDE user. I use IntelliJ

**Comment by brenuart** [comment]
Hmmm... The documentation says that `cacheFrom` is "_**List** of image names to use as cache sources_".
When defined inline in the plugin configuration as an XML element, this should give something like:
```
<image>
  <build>
    <cacheFrom>
      <cache>image1:latest</cache>
      <cache>image2:latest</cache>
    </cacheFrom>
   ...
```

When defined as a property, we should probably do something like:
```
docker.cacheFrom.1 = image1:latest
docker.cacheFrom.2 = image2:latest
```
If I understand correctly, the documentation at https://dmp.fabric8.io/#property-configuration should refer use `docker.cacheFrom.idx` to refer to this property. Correct?

If I'm right - this explains why the test fails...

**Comment by brenuart** [comment]
Any feedback about this PR ?

**Comment by rohanKanojia** [comment]
Do you think we need to update documentation too https://dmp.fabric8.io/#property-configuration for cacheFrom?

**Comment by brenuart** [comment]
I have updated the documentation. I also created a new section dedicated to List Properties instead of repeating how they work for every property of this type.

**Comment by brenuart** [comment]
Done! Is everything ok now ?

File: src/main/asciidoc/inc/external/_property_configuration.adoc [hunk 1]
```diff
@@ -344,6 +354,22 @@ configuration are both used for running container and building
...
+[source,xml]
+----
+<docker.ports.1>80<docker.ports.1>
+<docker.ports.2>8080<docker.ports.2>
+
+<docker.tags.jenkins>${BUILD_TIMESTAMP}</docker.tags.jenkins>
+<docker.tags.current>latest</docker.tags.current>
+----
+
+The _idx_ suffixe defines the order of copying. Entries which _idx_ is an integer number are copied first in ascending order by the parsed value of _idx_. Entries which _idx_ is not an integer number are copied second in ascending alphabetical order by _idx_.
```
Comment (rohanKanojia) [inline src/main/asciidoc/inc/external/_property_configuration.adoc:371]:
```suggestion
The _idx_ suffix defines the order of copying. Entries which _idx_ is an integer number are copied first in ascending order by the parsed value of _idx_. Entries which _idx_ is not an integer number are copied second in ascending alphabetical order by _idx_.
```

**Review by rohanKanojia — APPROVED** [review]

**Comment by rohanKanojia** [comment]
@brenuart : Could you please add a line to [doc/changelog.md](https://github.com/fabric8io/docker-maven-plugin/blob/master/doc/changelog.md) regarding this change? Are we changing cacheFrom behavior from old releases? If yes, Shall we add a note about that in doc/changelog?

**Comment by brenuart** [comment]
Done. I added an entry in the `changelog.md` with a warning about the fact that `docker.cacheFrom` alone without an index suffix won't work anymore (as far as it ever worked correctly.

**Comment by brenuart** [comment]
Do you want me to squash and rebase before you merge the PR ?

**Pull request merged**